packages/multichain-account-service/src/MultichainAccountService.ts

```diff
@@ -38,17 +40,28 @@ function selectKeyringControllerKeyrings(state: KeyringControllerState) {
   return state.keyrings;
 }
 
+/** Reverse mapping object used to map account IDs and their wallet/multichain account. */
+type AccountReverseMapping<Account extends Bip44Account<KeyringAccount>> = {
+  wallet: MultichainAccountWallet<Account>;
+  multichainAccount: MultichainAccount<Account>;
+};
+
 /**
  * Service to expose multichain accounts capabilities.
  */
 export class MultichainAccountService {
   readonly #messenger: MultichainAccountServiceMessenger;
 
-  readonly #providers: AccountProvider<InternalAccount>[];
+  readonly #providers: AccountProvider<Bip44Account<InternalAccount>>[];
 
   readonly #wallets: Map<
     MultichainAccountWalletId,
-    MultichainAccountWallet<InternalAccount>
+    MultichainAccountWallet<Bip44Account<InternalAccount>>
+  >;
+
+  readonly #reverse: Map<
+    InternalAccount['id'],
+    AccountReverseMapping<Bip44Account<InternalAccount>>
```

Comment:
```suggestion
  readonly #accountIdToContext: Map<
    Bip44<InternalAccount>['id'],
    AccountReverseMapping<Bip44Account<InternalAccount>>
```

Can we be more descriptive here with the type and do `Bip44Account<InternalAccount>['id']` instead?